torch/distributed/rpc/internal.py

```diff
@@ -259,3 +303,17 @@ def _start_record_function(exec_type, func_name, current_worker_name, dest_worke
 
 PythonUDF = collections.namedtuple("PythonUDF", ["func", "args", "kwargs"])
 RemoteException = collections.namedtuple("RemoteException", ["msg", "exception_type"])
+# FIXME: Directly referencing `dist.nn._REMOTE_MODULE_PICKLED_ATTRIBUTES` here can cause
+# "AttributeError: module 'torch.distributed' has no attribute 'nn'".
+# RemoteModule = collections.namedtuple("RemoteModule", dist.nn._REMOTE_MODULE_PICKLED_ATTRIBUTES)
```

Comment:
This looks like a hack. Besides, IIUC, `RemoteModule` is a layer on top of RPC/RRef. Referencing `RemoteModule` in RPC source code looks a bit weird. 

Is it possible to let RPC pickler expose an private API for other modules to register their reducers. And RemoteModule call that API to add itself?

Reply:
> This looks like a hack. Besides, IIUC, `RemoteModule` is a layer on top of RPC/RRef. Referencing `RemoteModule` in RPC source code looks a bit weird.

This looks like a hack. 
Oh, this indeed a hack, by resolving the suggestion from @lw and avoid hard-coding the fields in reducer/receiver side.
 
> Is it possible to let RPC pickler expose a private API for other modules to register their reducers. And RemoteModule call that API to add itself?
Good idea. Let me try that this approach.

Reply:
Thanks for the suggestion! Created a private API to register reducers. Now `RemoteModule` is not imported to `internal.py`.